### PR TITLE
Add how to release guide

### PR DESCRIPTION
**Pull request by epeters3**

### CODE REVIEW

File: HOW_TO_RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# How To Release the `metalearn` Package
+
+### Testing
+
+1. Using a clean environment, do a final test of the current state of the `develop` branch. Make sure before tests are run to delete or move the `metalearn` directory so the tests are run against the installed version of `metalearn` and not the local files.
+1. Test using the develop branch of the `metalearn` package with our `d3m_primitives` package, to make sure the APIs are compatible.
```
Comment (bjschoenfeld) [inline HOW_TO_RELEASE.md]: Should we link the `d3m_primitives` package?

**Comment by bjschoenfeld** [comment]
Maybe the numbering should be all 1's.

File: HOW_TO_RELEASE.md (hunk 1 above)
Comment (epeters3) [reply HOW_TO_RELEASE.md]: That's a good idea, I'll do that.

**Comment by epeters3** [comment]
So `.md` file parsers typically always parse the `1.` prefix to a line and assign the correct ordered list number to them when rendering. So when viewing this file on github, it should have the correct numberings. When viewing the file raw though, each line will still only have the `1.`, so it should be ok.

File: HOW_TO_RELEASE.md (hunk 1 above)
Comment (epeters3) [reply HOW_TO_RELEASE.md]: Finished.

**Pull request merged**